Add support for private_key_file and private_key_file_pwd params

bennylu2 opened the pull request:
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #1565

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [x] I am modifying authorization mechanisms
   - [ ] I am adding new credentials
   - [ ] I am modifying OCSP code
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

Adding support for specifying a private key file and password for authenticating with Snowflake. Connector already supports `private_key` but it must be provided as bytes. The JDBC driver supports these as the parameters [private_key_file](https://docs.snowflake.com/en/developer-guide/jdbc/jdbc-parameters#private-key-file) and `private_key_file_pwd` but is missing from the `snowflake-connector-python` project namely from `auth/keypair.py`

bennylu2 commented:
I have read the CLA Document and I hereby sign the CLA

bennylu2 commented on src/snowflake/connector/auth/keypair.py:
```diff
@@ -19,6 +19,7 @@
     Encoding,
     PublicFormat,
     load_der_private_key,
+    load_pem_private_key
```
```suggestion
    load_pem_private_key,
```

sfc-gh-mkeller commented:
I'd prefer if `src/snowflake/connector/auth/keypair.py` was left untouched, and we introduced all the logic in the `connection.py`.
We can just read in the key and decrypt it and then treat it like it was provided to the connection as straight `bytes`.

bennylu2 commented:
> I'd prefer if `src/snowflake/connector/auth/keypair.py` was left untouched, and we introduced all the logic in the `connection.py`. We can just read in the key and decrypt it and then treat it like it was provided to the connection as straight `bytes`.

Refactored, is this what you were thinking?

sfc-gh-mkeller commented:
Yes, that looks wonderful! Please add a changelog entry and a unit test for the loading the private key. I'd love to accept this afterwards!

bennylu2 commented:
> Yes, that looks wonderful! Please add a changelog entry and a unit test for the loading the private key. I'd love to accept this afterwards!

Done! I noticed there's not really tests for connection.py so I made the loading private key a function to write a unit test. Lmk if there's something I missed

bennylu2 commented:
Could the workflows be enabled please? Would be good to see if there's any failing workflows/checks

sfc-gh-mkeller commented:
Merged through #1808 
❤️ @bennylu2
